### PR TITLE
Added release process documentation

### LINKED ISSUE #39: initial release process docs for ccxcon
based on LORE docs

### PR DESCRIPTION
**Pull request by pwilkins**
Added a document to describe the release process step-by-step.
The document is based on a similar document for LORE.

Fixes #39

@bdero

### CODE REVIEW

**Comment by justinabrahms** [comment]
I'm a little surprised that this isn't a script. Is there a reason this is multiple manual steps, rather than an automated process that's kicked off by humans (e.g. something like `make-release --minor` or similar)

**Comment by pwilkins** [comment]
I'm all in favor of a script if it could be written.  The fact that the process is difficult to do manually is either evidence that a script is needed or that a script wouldn't manage all decision points.  (Or both.)

File: docs/release_process.rst [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    git pull
+    git flow release start 0.10.0
+    vi ccxcon/settings.py # change version number
+    git-release-notes v0.9.0..master util/release_notes_rst.ejs
+    vi RELEASE.rst # paste results of prior command at top of file, updating the release number
+    git add RELEASE.rst ccxcon/settings.py
+    git commit -m "Release 0.10.0"
+    git flow release publish 0.10.0
+    # Verify that the Travis-ci build succeeds
+    git-release-notes v0.9.0..master util/release_notes.ejs
```
Comment (justinabrahms) [inline docs/release_process.rst:29]: Why do we do this twice? Once here and once on line 23?

Comment (pwilkins) [inline docs/release_process.rst:29]: These are two different scripts, though you might miss that at first glance.  One creates a reStructuredText document that is included in our ReadTheDocs documentation, and the other creates checkboxes for the `release` PR description.

**Comment by justinabrahms** [comment]
Filed a ticket for my desire to have a more automated release in #43. Happy to work on that when we're in a better place from a scheduling point of view.

**Comment by bdero** [comment]
Yeah, early on we talked about writing a script that would automate all of those steps and open PRs with the github API (can probably just use `hub` for simplicity). In fact, I'm pretty sure Carson or I made a backlog entry for it way back when we were using trello.

**Comment by bdero** [comment]
There is a version listed in both `ccxcon/settings.py` (because there is a test that checks to make sure we're using valid semver) as well as in `setup.py`.

I'll make a issue about either removing `setup.py` or placing the version in a `VERSION` text file and loading it from disk.

You may want to add a step in the meantime for changing the version in `setup.py` as well.. but other than that, I say LGTM. :+1:

**Comment by pdpinch** [comment]
This is good for now. After we get some experience under our belts, I would expect this doc also to address coordinating releases of CCXcon and Teachers Portal.

**Comment by pdpinch** [comment]
@bdero what's `hub` ?

**Comment by bdero** [comment]
`hub` is a nice little github CLI: https://github.com/github/hub

A real time-saver for personal use!

**Comment by justinabrahms** [comment]
Can we merge this, @pwilkins ?

**Pull request merged**